azerothcore/azerothcore-wotlk
# fix(Core/Spells): Cooldowns:

Reworked spell category cooldowns.
Implemented category cooldowns for pets.
Properly shows pet spell cooldowns in player's UI.
Corrected pet spell cooldowns with infinity duration.
Do not add/remove infinity spell cooldown on aura apply/remove if casted by item.
Fixes #5263.



## Issues Addressed:

- Closes #5263
- Closes https://github.com/chromiecraft/chromiecraft/issues/386
- Closes https://github.com/chromiecraft/chromiecraft/issues/1350

## Tests Performed:

- Tested ingame with many spells and pets.

## How to Test the Changes:

```
Properly shows pet spell cooldowns in player's UI.
Corrected pet spell cooldowns with infinity duration.
```
1. Login on warlock and summon imp
2. Cast "PhaseShift"
3. Logout
4. Login
5. See that "PhaseShift" is still on inifity duration (it's active).
6. Remove aura by clicking on pet bar
7. You should see cooldown counting.

**For all other problems described here - see #5263**


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
